llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -8257,11 +8260,11 @@ VPRecipeBuilder::tryToCreatePartialReduction(VPInstruction *Reduction,
     ReductionOpcode = Instruction::Add;
   }
 
-  VPValue *Cond = nullptr;
-  if (CM.blockNeedsPredicationForAnyReason(ReductionI->getParent()))
-    Cond = getBlockInMask(Builder.getInsertBlock());
-  return new VPPartialReductionRecipe(ReductionOpcode, Accumulator, BinOp, Cond,
-                                      ScaleFactor, ReductionI);
+  assert(!CM.blockNeedsPredicationForAnyReason(ReductionI->getParent()) &&
+         "Should not create partial reductions for predicated blocks");
+  return new VPPartialReductionRecipe(ReductionOpcode, Accumulator, BinOp,
+                                      /*Cond=*/nullptr, ScaleFactor,
```

Comment:
Should we remove Cond from the `VPPartialReductionRecipe` constructor and just always pass nullptr to `VPReductionRecipe`?